xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-ui/src/main/resources/XWiki/Authentication/Administration.xml

```diff
@@ -36,7 +36,75 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{velocity}}
+  <content>(% id='HAuthService' %)
+== {{translation key="security.authservice.admin.title"/}}
+
+{{velocity}}
+#if ($request.setauthservice &amp;&amp; $request.authServiceId)
+  #if ($services.csrf.isTokenValid($request.getParameter('form_token')))
+    $services.security.authService.setAuthService($request.authServiceId)
+  #end
+#end
+{{/velocity}}
+
+{{velocity}}
+#set ($authService = $services.security.authService.getAuthService())
+#set ($configuredAuthClass = $services.security.authService.getConfiguredAuthClass())
+#if ($services.security.authService.isAuthServiceComponent())
+  #set ($authServiceType = 'component')
+#elseif ($configuredAuthClass &amp;&amp; ($authService.class.name == $configuredAuthClass || $authService.class.name == 'com.xpn.xwiki.user.impl.xwiki.XWikiAuthServiceImpl'))
+  #set ($authServiceType = 'class')
+#else
+  #set ($authServiceType = 'other')
+#end
+
+{{box}}
+  #set ($authService = $services.security.authService.getAuthService())
+  #set ($authServiceName = $services.localization.get("security.authservice.service.${authService.id}.name"))
+  #if ($authServiceName)
+    **{{translation key="security.authservice.service.${authService.id}.name"/}}** (//$authService.class.name//)
+    #set ($authServiceDescription = $services.localization.get("security.authservice.service.${authService.id}.description"))
+    #if ($authServiceDescription)
+      {{translation key="security.authservice.service.${authService.id}.description"/}}
+    #end
+  #else
+    $authService.class.name
+  #end
+{{/box}}
+
+#if ($authServiceType == 'component')
+  #set ($authServices = $services.security.authService.getAuthServices())
+  #if ($authServices.size() &gt; 1)
+    {{html}}
+      &lt;form action="$xwiki.relativeRequestURL" method="post"&gt;
+        &lt;input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" /&gt;;
```

Comment:
No need to xml escape here?

Reply:
Superflous `;`?
```suggestion
      &lt;form action="$xwiki.relativeRequestURL" method="post"&gt;
        &lt;input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" /&gt;
```

Reply:
> No need to xml escape here?

Not with the current token format, I guess we could argue that it's more future-proof to escape anyway but note that our code is full of that exact same code.

> Superflous ;?

Looks like it indeed, not sure how it ended up there.